scripts/build/README.md

```diff
@@ -0,0 +1,58 @@
+# CHIP automated builds scripts
+
+CHIP compilation is generally split into two steps
+
+1. Generate ninja/makefile for out-of-source builds
+2. Compilation using ninja/makefiles
+
+## Building manually
+
+Manual building is generally platform-dependent. All build steps would require a
+bootstrapped environment (loads a pigweed build environment) and will then be
+followed by platform-specific instructions.
+
+The file BUILDING.md describes general requirements and examples. Typical usage
+is:
+
+```
+source scripts/activate
+gn gen out/host
+ninja -C out/host
+```
+
+## Unified build script
+
+The script `chipbuild.py` provides a single entry point for generating and
+executing the build.
+
+Build environment _MUST_ be properly configured for chipbuild to succeed. For
```

Comment:
I think most of the utility of having such a script won't be realized until we fix this limitation.

Reply:
This was one of the reasons I though python can work better - it can validate/give better debug information for when things fail.

There are tools having `doctor` instructions that validate the enviroment or we can make the tool try to install dependencies from packages/docker. How do do this is non-obvious. We can build anything that pigweed downloads or we have in third_party, however external SDKS (thinkin esp or nrfconnect) need to somehow be available.